tools/codegen/cmd/featuregate-test-analyzer.go

```diff
@@ -543,7 +543,17 @@ func getLatestRelease() (string, error) {
 		return "", fmt.Errorf("no releases found")
 	}
 
-	return result.Releases[0], nil
+	// if its not main branch, then use the ENV var to determine the release version
+	currentRelease := os.Getenv("PULL_BASE_REF")
+	if len(currentRelease) == 0 {
+		return "", fmt.Errorf("unable to fetch the release version")
+	}
+	if strings.Contains(currentRelease, "release-") {
+		// example: release-4.18, release-4.17
+		return strings.TrimPrefix(currentRelease, "release-"), nil
+	}
```

Comment:
This should come first, before we even call sippy

Reply:
done.